modules/rostests/apitests/shell32/CommandLineToArgvW.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Test for CommandLineToArgvW
+ * COPYRIGHT:   Copyright 2023 Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
+ */
+
+#include "shelltest.h"
+#include <atlstr.h>
+
+static VOID
+DoEntry(INT lineno, LPCWSTR cmdline, INT expected_argc, ...)
+{
+    va_list va;
+    va_start(va, expected_argc);
+
+    INT real_argc;
+    LPWSTR *real_argv = CommandLineToArgvW(cmdline, &real_argc);
+
+    INT common = min(expected_argc, real_argc);
+    for (INT i = 1; i < common; ++i)
+    {
+        CStringA str1 = wine_dbgstr_w(va_arg(va, LPCWSTR));
+        CStringA str2 = wine_dbgstr_w(real_argv[i]);
+        ok(str1 == str2, "Line %d: %s != %s\n", lineno, (LPCSTR)str1, (LPCSTR)str2);
+    }
+
+    INT diff = abs(expected_argc - real_argc);
+    while (diff-- > 0)
+    {
+        ok(expected_argc == real_argc,
+           "Line %d: expected_argc %d and real_argc %d are different\n",
+           lineno, expected_argc, real_argc);
+    }
```

Comment:
Probably easier to just do something like:
`ok(expected_argc == real_argc, "Line %d: expected_argc %d and real_argc %d are different\n", lineno, expected_argc, real_argc);`

Reply:
To count the number of failures correctly, the original code is better.